name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/AbsolutePerformancePieChart.java

```diff
@@ -0,0 +1,334 @@
+        var dataSeriesSet = new DataSeriesSet(client, preferences, UseCase.STATEMENT_OF_ASSETS);
+
+        countAssets(wrapper, dataSeriesSet);
+
+        countAccounts(wrapper, dataSeriesSet);
+
+        if (wrapper.invested <= 0)
+            return;
+        
+        if (wrapper.counter > 1)
+        {
+            position = null;
+        }
+
+        wrapper.delta = Math.max(0, wrapper.delta);
+
+        var total = wrapper.invested + wrapper.delta;
+
+
+        values.add(wrapper.invested / Values.Amount.divider());
+        labels.add(Messages.LabelAbsoluteInvestedCapital);
+
+        NodeData data = new NodeData();
+        data.label = Messages.LabelAbsoluteInvestedCapital;
+        data.percentageString = Values.Percent2.format((wrapper.invested / (double) total));
+        data.value = Values.Money.format(Money.of(snapshot.getCurrencyCode(), wrapper.invested));
+        id2nodeData.put(Messages.LabelAbsoluteInvestedCapital, data);
+
+
+        values.add(wrapper.delta / Values.Amount.divider());
+        labels.add(Messages.LabelAbsoluteDelta);
+
+        data = new NodeData();
+        data.label = Messages.LabelAbsoluteDelta;
+        data.percentageString = Values.Percent2.format((wrapper.delta / (double) total));
+        data.value = Values.Money.format(Money.of(snapshot.getCurrencyCode(), wrapper.delta));
+        id2nodeData.put(Messages.LabelAbsoluteDelta, data);
+
+
+        ICircularSeries<?> circularSeries = (ICircularSeries<?>) chart.getSeriesSet()
+                        .getSeries(Messages.ColumnAbsolutePerformance_MenuLabel);
+        if (circularSeries == null)
+        {
+            circularSeries = createSeries(values, labels);
+        }
+        else
+        {
+            // refresh values only for smoother update
+            ListIterator<String> iter = labels.listIterator();
+            while (iter.hasNext())
+            {
+                int idx = iter.nextIndex();
+                String label = iter.next();
+                Node node = circularSeries.getNodeById(label);
+                if (node != null)
+                {
+                    node.setValue(values.get(idx));
+                }
+            }
+
+        }
+
+        setColors(circularSeries, values.size(), wrapper.delta > 0);
+
+        chart.updateAngleBounds();
+        chart.redraw();
+    }
+
+    private void countAssets(InvestedWrapper wrapper, DataSeriesSet dataSeriesSet)
+    {
+        snapshot.getAssetPositions() //
+                        .filter(p -> p.getValuation().getAmount() > 0) //
+                        .forEach(p -> {
+                            String nodeId = p.getInvestmentVehicle().getUUID();
+                            
+                            var dataSeries = dataSeriesSet.getAvailableSeries().stream()
+                                            .filter(ds -> ds.getType() != Type.ACCOUNT && ds.getUUID().contains(nodeId))
+                                            .findAny().orElse(null);
```

Comment:
You use the ClientSnapshot and then lookup the data series (meant for the charts).

Can't we use the ClientPerformanceSnapshot? It has the list of positions plus all the components that have been held in-between.